File: ios/WebyarNative/Sources/Core/Models/SayNowVoice.swift
import Foundation

/// Whose voice the AI writes in when it delivers an operator's message.
///
/// The distinction is the visitor's, not the operator's: the same sentence
/// reads differently coming from a person on the team than from the
/// assistant, and on a thread the AI is already answering the operator is
/// choosing which of those the visitor should believe they are talking to.
enum SayNowVoice: String, CaseIterable, Identifiable, Sendable {
    /// Written as one of the humans on the team.
    case specialist
    /// Written as the assistant, openly.
    case assistant

    var id: String { rawValue }

    var icon: String {
        switch self {
        case .specialist: "person.wave.2.fill"
        case .assistant: "sparkles"
        }
    }

    func title(_ l: Language) -> String {
        switch self {
        case .specialist: Str.sayNowVoiceSpecialist(l)
        case .assistant: Str.sayNowVoiceAssistant(l)
        }
    }
}

File: android/app/src/main/kotlin/com/webyar/operator/core/model/SayNowVoice.kt
package com.webyar.operator.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Whose voice the AI writes in when it delivers an operator's message.
 *
 * The distinction is the visitor's, not the operator's: the same sentence
 * reads differently coming from a person on the team than from the assistant,
 * and on a thread the AI is already answering, the operator is choosing which
 * of those the visitor should believe they are talking to.
 */
@Serializable
enum class SayNowVoice {
    /** Written as one of the humans on the team. */
    @SerialName("specialist") SPECIALIST,

    /** Written as the assistant, openly. */
    @SerialName("assistant") ASSISTANT;

    val wireValue: String
        get() = when (this) {
            SPECIALIST -> "specialist"
            ASSISTANT -> "assistant"
        }
}
